library/src/main/java/com/wultra/android/mtokensdk/operation/TOTPUtils.kt

```diff
@@ -0,0 +1,74 @@
+package com.wultra.android.mtokensdk.operation
+
+import android.net.Uri
+import android.util.Base64
+import com.google.gson.Gson
+import com.google.gson.annotations.SerializedName
+import com.wultra.android.mtokensdk.common.Logger
+
+/**
+ * Utility class used for handling TOTP
+ */
+class TOTPUtils {
+
+    /** Data payload which is returned from JWT parser */
+    data class OperationTOTPData(
+
+        /** The ID of the operations associated with the TOTP */
+        val totp: String,
+
+        /** The actual Time-based one time password */
+        @SerializedName("oid")
+        val operationId: String
+    )
+
+    companion object {
+
+        /** Method accepts deeplink Uri and returns payload data or null */
+        fun parseDeeplink(uri: Uri?): OperationTOTPData? {
+            val query = uri?.query ?: return null
+            val queryItems = query.split("&").associate {
+                val (key, value) = it.split("=")
+                key to value
+            }
+
+            if ("code" in queryItems) {
```

Comment:
I think this could be simplified: `queryItems["code"]?.let { parseJWT(it) }` + some logging